### ISSUE TITLE
Sphere Tessellation in OpenGL and F#

### ISSUE THREAD

**Issue by gileoo** [uncodeference/unmeetup#11]
- **Marcel Ritter**
- **Beginner**
- **20 Minutes**
- **OpenGL, Tessellation, F#, Multiplatform**

# \<Tessellated Spheres in OpenGL and F#>

### One-sentence summary
Presentation of the concept and the implementation of an interactive viewer for point clouds by generating screen-adaptive tessellated spheres using a minimal code written in F# using OpenGL and OpenTK. 

### What's the format — is it a case study, a live coding session, a workshop or something else?
Frontal talk with demo and discussion about a first prototype code.

### Tell us more about the talk
Stemming from a computer graphics background, I was interested in stirring OpenGL via F# and compile to a multi-platform application. Based on the work on a viewer for molecular dynamics simulations written in C++, OpenGL and Imgui, the atoms viewing part was transformed into a point cloud viewer. Concepts from the F# world were applied with the code transformation, e.g. immutability, pipeline operation, record types for objects/states, state transition functions., ... The minimal code now compiles and runs in Windows and Linux, and provides helpers for OpenGL programming,  to be discussed in the session. Also, the concept of the geometry generation is presented, being independent from the host programming language.

### A few words about yourself
**Academic Background:** BSc in Civil Engineering, Msc in Compter Science, PhD candidate in Computer Science at the Interactive Graphics and Simulation Group at the University of Innsbruck specializing in scientific visualization and computational geometry.  
**Industrial Background:**
Game development at Clockstone, Head of Software Development at Airborne Hydromapping (AHM).

### Would you be willing to have a Q/A session after the talk?
Yes.

### How can we find you on social media?
www.marcel-ritter.com
Ping me there for skype, telegram, and fb.

**Comment by bemayr** [comment]
Thanks Marcel for proposing this talk! 😍
F# again, the functional programming community in Innsbruck grows and grows, we are really looking forward to your talk. And should probably introduce a **n-times a row speaker award** for you! 😉

**Comment by gileoo** [comment]
Yeah, thnx. But I don't want to over-do things. You can just kick me out, when its getting too much. :yum:
